shipyard-controller/db/mongodb_connection.go

```diff
@@ -12,11 +12,20 @@ import (
 
 var mutex = &sync.Mutex{}
 
+var mongoDBConnectionInstance *MongoDBConnection
+
 // MongoDBConnection takes care of establishing a connection to the mongodb
 type MongoDBConnection struct {
 	Client *mongo.Client
 }
 
+func GetMongoDBConnectionInstance() *MongoDBConnection {
```

Comment:
This is not thread-safe.
I would propose something like this:
```
var mongoConnectionOnce sync.Once
...
func GetMongoDBConnectionInstance() *MongoDBConnection {
	mongoConnectionOnce.Do(func() {
		mongoDBConnectionInstance = &MongoDBConnection{}
		mongoDBConnectionInstance.EnsureDBConnection()
	})
	return mongoDBConnectionInstance
}
```
We could also skip the calls to `EnsureDBConnection()` in all the repos.